### ISSUE TITLE
Instrument View: Error unwrapping surface for some instruments

### ISSUE THREAD

**Issue by martyngigg** [mantidproject/mantid#31586]
### Expected behavior

Instrument view shows unwrapped view of instrument.

### Actual behavior

An error is generated for some IDFs, e.g CRISP & INTER.

This is regression as the same steps work in version 6.0.

### Steps to reproduce the behavior

* Start Workbench
* Run `LoadEmptyInstrument` and set `InstrumentName` to `CRISP` and set a workspace name
* Click show instrument on the workspace
* Click the box showing "Full 3D" and switch it to one of the other flat projections like Cylinder Y
* An error dialog is generated.

### Platforms affected

All

**Comment by DannyHindson** [comment]
I've just tried this out - I see the problem in the latest code but I also see it in v6.0. So doesn't look like a regression to me:

![image](https://user-images.githubusercontent.com/56295817/119647505-415c6980-be18-11eb-9a5f-3dc32cfd0fba.png)

**Comment by martyngigg** [comment]
Oh. I wonder why it seemed to work for me in 6.0.

Would you still mind having a look at how large/complex a fix might be to see if it's safe to include?

**Comment by DannyHindson** [comment]
I've had a look into this. I think this can be fixed with a couple of one line changes:

- I think the warning can simply be removed. It's created in a function that tries to spot if a bank of detectors has been split into two when it's unwrapped into a rectangle using a cylindrical projection. For a bank of detectors with width 1 (like in CRISP, INTER) the function doesn't need to do anything
- The values u_min and u_max are currently set to 0 and 1 respectively immediately after the warning. Since u is measured in radians, the value of u_max=1 doesn't make much sense. I suggest setting u_max=0. I've tried this and it works OK

```
void RotationSurface::findAndCorrectUGap() {
  double period = uPeriod();
  if (period == 0.0)
    return;
  const int nbins = 1000;
  std::vector<bool> ubins(nbins);
  double bin_width = fabs(m_u_max - m_u_min) / (nbins - 1);
  if (bin_width == 0.0) {
    QApplication::setOverrideCursor(QCursor(Qt::ArrowCursor));
    QMessageBox::warning(
        nullptr, tr("Mantid - Instrument view warning"),
        tr("Rotation surface: failed to build unwrapped surface"));
    QApplication::restoreOverrideCursor();
    m_u_min = 0.0;
    m_u_max = 1.0;
    return;
  }
```

A few commits that I used:
findAndCorrectUGap added: https://github.com/mantidproject/mantid/commit/dd96095e6c0ee7e965bf26e48702f5556152993f
exception added: https://github.com/mantidproject/mantid/commit/82b17a1c7b704c352a7abf37384818fcc15749e3
downgraded to a warning: https://github.com/mantidproject/mantid/commit/a3b891397fc7c689a31e97e19ece8075549634c2

**Comment by martyngigg** [comment]
Thanks for investigating this.

> I've had a look into this. I think this can be fixed with a couple of one line changes:
> 
>     * I think the warning can simply be removed. It's created in a function that tries to spot if a bank of detectors has been split into two when it's unwrapped into a rectangle using a cylindrical projection. For a bank of detectors with width 1 (like in CRISP, INTER) the function doesn't need to do anything

Agreed. I suppose we could spit out something out at debug if we think it's useful to know that but not required I don't think.

>     * The values u_min and u_max are currently set to 0 and 1 respectively immediately after the warning. Since u is measured in radians, the value of u_max=1 doesn't make much sense. I suggest setting u_max=0. I've tried this and it works OK

Agreed. This would be more correct in line with what the code is saying from the `bin_width` calculation.

**Comment by martyngigg** [comment]
Fixed by #31590
